src/logging.rs

```diff
@@ -82,16 +83,18 @@ fn add_field_and_payload(data: &mut String, field: &str, payload: &str) {
         data.push_str(&field_payload);
     }
 }
-
-/// Print a message to the journal with the given priority.
-///
+/// Send a message to the local systemd journal. 
+/// vars is a map of journald fields to values.
 /// This uses the default systemd-journald socket.
-pub fn journal_print(priority: Priority, msg: &str) -> Result<()> {
+fn journal_send(priority: Priority, msg: &str,vars: &mut HashMap<&str, &str>) -> Result<()> {
```

Comment:
This would be a `pub fn`.

Also, no reason for `vars` to be marked as `mut`, I think.